plugins/flytekit-kf-tensorflow/flytekitplugins/kftensorflow/models/chief.py

```diff
@@ -0,0 +1,58 @@
+from typing import Optional
+
+from flyteidl.plugins.kubeflow import tensorflow_pb2 as tensorflow_task
+
+from flytekit.models import common, task
+
+from .restart_policy import RestartPolicy
+
+
+class Chief(common.FlyteIdlEntity):
+    """
+    Configuration for a chief replica group in a TFJob.
+
+    :param replicas: Number of replicas in the group. This should be 1 or 0. If 0, the chief will be elected from the worker group.
+    :param image: Optional image to use for the pods of the group
+    :param resources: Optional resources to use for the pods of the group
+    :param restart_policy: Optional restart policy to use for the pods of the group
+    """
+
+    def __init__(
+        self,
+        replicas: int,
+        image: Optional[str] = None,
+        resources: Optional[task.Resources] = None,
+        restart_policy: Optional[RestartPolicy] = None,
+    ):
+        if replicas != 0 and replicas != 1:
+            raise ValueError(
+                f"TFJob chief group needs to have either one replica or no replica(one worker will be elected as chief), but {replicas} have been specified."
+            )
+        self._replicas = replicas
+        self._image = image
+        self._resources = resources
+        self._restart_policy = restart_policy
+
+    @property
+    def image(self) -> Optional[str]:
+        return self._image
+
+    @property
+    def resources(self) -> Optional[task.Resources]:
+        return self._resources
+
+    @property
+    def replicas(self) -> Optional[int]:
+        return self._replicas
+
+    @property
+    def restart_policy(self) -> Optional[RestartPolicy]:
+        return self._restart_policy
+
+    def to_flyte_idl(self) -> tensorflow_task.DistributedTensorflowTrainingReplicaSpec:
+        return tensorflow_task.DistributedTensorflowTrainingReplicaSpec(
+            replicas=self.replicas,
+            image=self.image,
+            resources=self.resources.to_flyte_idl() if self.resources else None,
+            restart_policy=self.restart_policy.value if self.restart_policy else None,
+        )
```

Comment:
@yubofredwang Thanks for working on this
We plan to remove all the model files in flytekit, could you just use it inline [here](https://github.com/flyteorg/flytekit/pull/1627/files#diff-975a133e6c3847128c36fabd5b98f85aded0bed060412643ef44624707e179a6R117)

Reply:
Do you still want to keep the model class or it should be moved to inner logic to get_custom?

Reply:
we don't need model class anymore, just move to inner logic to get_custom

Reply:
Updated, please verify if this is something you want.